examples/banking-transaction-system/processor/main.bal

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) 2025, WSO2 LLC. (http://www.wso2.org).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/io;
+import ballerinax/kafka;
+
+// Schema Registry Configuration
+configurable string baseUrl = ?;
+configurable map<anydata> & readonly originals = ?;
+
+// Kafka Configuration
+configurable string bootstrapServers = ?;
+configurable string kafkaUsername = ?;
+configurable string kafkaPassword = ?;
+configurable string topicName = ?;
+configurable string groupId = ?;
+
+// Transaction Status Enum
+public enum TransactionStatus {
+    PENDING,
+    COMPLETED,
+    FAILED,
+    CANCELLED
+}
+
+// Transaction Type Enum
+public enum TransactionType {
+    DEPOSIT,
+    WITHDRAWAL,
+    TRANSFER
+}
+
+// Transaction Record
+type Transaction record {
+    string transaction_id;
+    string account_number;
+    TransactionType transaction_type;
+    float amount;
+    string currency;
+    string timestamp;
+    TransactionStatus status;
+    string description;
+    string? recipient_account;
+};
+
+// Kafka Consumer Configuration
+listener kafka:Listener transactionListener = new (bootstrapServers, {
+    groupId: groupId,
+    topics: [topicName],
+    securityProtocol: kafka:PROTOCOL_SASL_SSL,
+    auth: {
+        mechanism: kafka:AUTH_SASL_PLAIN,
+        username: kafkaUsername,
+        password: kafkaPassword
+    },
+    schemaRegistryUrl: baseUrl,
+    schemaRegistryConfig: {
+        "baseUrl": baseUrl,
+        "originals": originals
+    },
+    valueDeserializerType: kafka:DES_AVRO,
+    offsetReset: kafka:OFFSET_RESET_EARLIEST
+});
+
+// Transaction Processing Service
+service on transactionListener {
+
+    remote function onConsumerRecord(anydata[] data) returns error? {
```

Comment:
Can't we databind values to the `Transaction` record directly ?

Reply:
We can. will update 👍

Reply:
Fixed with 6288e2896653aec6c844aa6a54332f03078023af